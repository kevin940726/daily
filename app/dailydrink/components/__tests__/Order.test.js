const { html } = require('../../../slack-block-kit');
const Order = require('../Order');

test('it renders <Order />', () => {
  const block = html`
    <${Order}
      order=${{
        orderID: 'orderID',
        orderName: 'orderName',
        userID: 'userID',
        size: 'l',
        ice: 'free',
        sugar: 'free',
        price: 50,
        ingredients: 'ingredients',
      }}
      messageID="messageID"
      isClosed=${false}
    />
  `;

  expect(block).toMatchSnapshot();
});

test('it renders <Order /> with closed', () => {
  const block = html`
    <${Order}
      order=${{
        orderID: 'orderID',
        orderName: 'orderName',
        userID: 'userID',
        size: 'l',
        ice: 'free',
        sugar: 'free',
        price: 50,
        ingredients: 'ingredients',
      }}
      messageID="messageID"
      isClosed
    />
  `;

  expect(block).toMatchSnapshot();
});
