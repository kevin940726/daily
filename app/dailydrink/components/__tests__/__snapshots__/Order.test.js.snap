// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders <Order /> 1`] = `
Object {
  "accessory": Object {
    "action_id": "messageID",
    "options": Array [
      Object {
        "text": Object {
          "emoji": true,
          "text": "Edit order",
          "type": "plain_text",
        },
        "value": "edit-order",
      },
      Object {
        "text": Object {
          "emoji": true,
          "text": "Remove order",
          "type": "plain_text",
        },
        "value": "remove-order",
      },
    ],
    "type": "overflow",
  },
  "block_id": "order-overflow-block-id_orderID",
  "text": Object {
    "text": "<@userID>: *orderName*  (*$50*)
L, Ice-free, Sugar-free, (ingredients)",
    "type": "mrkdwn",
  },
  "type": "section",
}
`;

exports[`it renders <Order /> with closed 1`] = `
Object {
  "accessory": undefined,
  "block_id": "order-overflow-block-id_orderID",
  "text": Object {
    "text": "<@userID>: *orderName*  (*$50*)
L, Ice-free, Sugar-free, (ingredients)",
    "type": "mrkdwn",
  },
  "type": "section",
}
`;
